src/features/dynamic-import.md

```diff
@@ -42,7 +42,7 @@ Here’s how to statically import and use the `./utils.mjs` module:
 ```
 
 :::note
-**Note:** The previous example uses the `.mjs` extension to signal that it’s a module rather than a regular script. On the web, file extensions don’t really matter, as long as the files are served with the correct MIME type (e.g. `text/javascript` for JavaScript files) in the `Content-Type` HTTP header. The `.mjs` extension is [especially useful](https://github.com/nodejs/node-eps/blob/master/002-es-modules.md#32-determining-if-source-is-an-es-module) on other platforms such as Node.js, where there’s no concept of MIME types or other mandatory hooks such as `type="module"` to determine whether something is a module or a regular script. We’re using the same extension here for consistency across platforms and to clearly make the distinction between modules and regular scripts.
+**Note:** The previous example uses the `.mjs` extension to signal that it’s a module rather than a regular script. On the web, file extensions don’t really matter, as long as the files are served with the correct MIME type (e.g. `text/javascript` for JavaScript files) in the `Content-Type` HTTP header.
```

Comment:
Why remove the last part of the paragraph? It’s all still relevant, AFAIK.

Reply:
https://nodejs.org/api/esm.html#esm_enabling shows that `.mjs` doesn't solve cases such as input code / pipe and there is a flag to simply enable that as ESM.

the whole CommonJS ecosystem grew and survived with everyone writing modules as `.js` without ever being confused.

Underlying that `.mjs` solves modules from regular scripts ambiguity, which was never the case in CommonJS, which also never existed on the Web, is misleading, as many developers never had that issue since you need to know what you are importing when importing something, you don't import files by accident.

Reply:
I'm not interested in rehashing the whole `.mjs` debate here. Again, let's keep this discussion on topic, and focus on the merits of this patch.

Reply:
It is in topic: the `.mjs` extension is **not** especially useful because it doesn't separate, as example, applications/programs by module.

The entry point of a web page is not a module that exports anything, it's an application/program that might use the native module system or not.

```html
<!doctype html>
<script type="module" src="app.js"></script>
```

If `app.js` is named `app.mjs` instead, reading the discussed argument, it means that `app.js` is a module that would export something.

That's 99% of the time not the case. `app.js` or any `main` / `index` / `entry` point on the Web is not a module per-se, so saying that `.mjs` solves ambiguity, when it has more ambiguity than a normal script, is misleading.

Entry points are scripts that might, or might not need, modules. The `.mjs` extension there becomes less semantic than a file that is a real module, exporting actually things.

Reply:
@WebReflection Despite [repeated](https://github.com/v8/v8.dev/pull/228#discussion_r317018300) [requests](https://github.com/v8/v8.dev/pull/228#discussion_r317019777) to the contrary, you continue to derail this conversation with strawman arguments and off-topic discussions. This behavior is inappropriate. Please note that [the Chromium Code of Conduct](https://v8.dev/terms#code-of-conduct) applies to this repository.

I'm trying to discuss the changes in @GeoffreyBooth's patch, and your actions are distracting from that. Let's try to work together and stay on topic.

Reply:
I am trying to explain why that sentence is misleading. Last attempt:

> The `.mjs` extension is especially useful on other platforms such as Node.js, where there’s no concept of MIME types or other mandatory hooks such as `type="module"`

Others platforms have the module flag, only NodeJS doesn't. SpiderMonkey and JSC can both use the `-m` flag, and the Web has `type="module"`.

If that point wants to be relevant, shouldn't it include which platform misses the flag to opt in for modules?

> We’re using the same extension here for consistency across platforms and to clearly make the distinction between modules and regular scripts.

All platforms have a flag to switch parsing goal, which is not solved by `.mjs` because Web needs mime type, SpiderMonkey still wants `-m` flag, and so does JSC.

So, what are these other platforms that would need `.mjs`, beside NodeJS?

If there was a list of other platforms it'd be a great point to keep, but as it is, it looks like a supporting argument for the sake of it.

Reply:
> Others platforms have the module flag, only NodeJS doesn't. SpiderMonkey and JSC can both use the `-m` flag, and the Web has `type="module"`.

JSC also recognizes `.mjs` in which case you don't need the `-m` flag. Same for `d8` where with `.mjs` you don't need the `--module` flag. Same for `deno`. Are you suggesting we include more non-Node.js examples?

> If that point wants to be relevant, shouldn't it include which platform misses the flag to opt in for modules?

I'm open to adding more non-Node.js examples, although I don't feel it's super valuable in this context. We're talking about a **brief side note** in a large article on JS modules; I don't think anyone reasonably expects it to be an exhaustive resource of all non-Web environments that run JavaScript.

> If there was a list of other platforms it'd be a great point to keep, but as it is, it looks like a supporting argument for the sake of it (aka: NodeJS is the only inconsistent here, not the other way around)

The article specifically talks about "platforms where there's no concept of MIME types or other mandatory hooks". This includes all JS engine binaries. I don't think it's relevant which of those special-case `.mjs`, since the point is that `.mjs` can help in these general cases (as it already does in `d8`/`jsc`/`deno`), without any downsides.

Reply:
> Are you suggesting we include more non-Node.js examples?

To keep it relevant, I think it should mention other non-NodeJS-related platforms that understands `.mjs` as is, even if these have also flags to opt in.

That would be IMO more informative than mentioning only NodeJS.

Reply:
Ok, let's update this section to use the link you suggested + include `d8` as an additional example.

```suggestion
:::note
**Note:** The previous example uses the `.mjs` extension to signal that it’s a module rather than a regular script. On the web, file extensions don’t really matter, as long as the files are served with the correct MIME type (e.g. `text/javascript` for JavaScript files) in the `Content-Type` HTTP header. The `.mjs` extension is [especially useful](https://nodejs.org/api/esm.html#esm_enabling) on other platforms such as Node.js and [`d8`](/docs/d8), where there’s no concept of MIME types or other mandatory hooks such as `type="module"` to determine whether something is a module or a regular script. We’re using the same extension here for consistency across platforms and to clearly make the distinction between modules and regular scripts.
```